server/src/main/java/org/opensearch/index/search/NestedHelper.java

```diff
@@ -155,6 +158,8 @@ public boolean mightMatchNonNestedDocs(Query query, String nestedPath) {
             return mightMatchNonNestedDocs(((PointRangeQuery) query).getField(), nestedPath);
         } else if (query instanceof IndexOrDocValuesQuery) {
             return mightMatchNonNestedDocs(((IndexOrDocValuesQuery) query).getIndexQuery(), nestedPath);
+        } else if (query instanceof ApproximateScoreQuery) {
+            return mightMatchNonNestedDocs(((ApproximateScoreQuery) query).getOriginalQuery(), nestedPath);
```

Comment:
Is there a way to decide or improve on which `instanceof` matching should be prioritized over others? Like more common queries should be higher up the matching list over others.

This doesn't has to be part of this change and can be taken as a `TODO` sometime later.